### PR TITLE
Add new hiffy based testing infrastructure

### PR DESCRIPTION
**Pull request by labbott**

### CODE REVIEW

**Comment by labbott** [comment]
See https://github.com/oxidecomputer/hubris/pull/1512 for the associated hubris changes

File: cmd/test/src/lib.rs [hunk 1]
```diff
@@ -227,441 +207,126 @@ impl fmt::Display for TestRunResult {
...
+        println!("{:#}", result);
+        writeln!(
+            out,
+            "==== Test {} result: {:?}",
+            test_name,
+            result.to_str_no_color()
+        )?;
+        writeln!(out, "==== Task state")?;
+
+        cmd_tasks::print_tasks(
```
Comment (bcantrill) [inline cmd/test/src/lib.rs:325]: I think that this is fine for now, but we may want to stick this into a different crate if we find we're using it a bunch....

**Review by bcantrill — CHANGES_REQUESTED** [review]
This looks great, and removes a bunch of gross code.  I would also like us to update the docs (which make reference to ITM -- and then have to explicitly apologize for its sporadic misbehavior), and then also bump the version.

**Review by bcantrill — APPROVED** [review]
Looks awesome -- thank you!

**Pull request merged**